core/src/main/java/cz/o2/proxima/scheme/AttributeValueAccessors.java

```diff
@@ -0,0 +1,323 @@
+  public static class StructureValueImpl implements StructureValue {
+    private final Map<String, Object> delegate;
+
+    private StructureValueImpl(Map<String, Object> delegate) {
+      this.delegate = delegate;
+    }
+
+    @Override
+    public Map<String, Object> value() {
+      return delegate;
+    }
+
+    @Override
+    @SuppressWarnings("unchecked")
+    public <V> V get(String name) {
+      return (V) delegate.get(name);
+    }
+
+    @Override
+    public int size() {
+      return delegate.size();
+    }
+
+    @Override
+    public boolean isEmpty() {
+      return delegate.isEmpty();
+    }
+
+    @Override
+    public boolean containsKey(Object key) {
+      return delegate.containsKey(key);
+    }
+
+    @Override
+    public boolean containsValue(Object value) {
+      return this.delegate.containsValue(value);
+    }
+
+    @Override
+    public Object get(Object key) {
+      return delegate.get(key);
+    }
+
+    @Override
+    public Object put(String key, Object value) {
+      return this.delegate.put(key, value);
+    }
+
+    @Override
+    public Object remove(Object key) {
+      return delegate.remove(key);
+    }
+
+    @Override
+    public void putAll(Map<? extends String, ?> m) {
+      delegate.putAll(m);
+    }
+
+    @Override
+    public void clear() {
+      delegate.clear();
+    }
+
+    @Override
+    public Set<String> keySet() {
+      return delegate.keySet();
+    }
+
+    @Override
+    public Collection<Object> values() {
+      return delegate.values();
+    }
+
+    @Override
+    public Set<Entry<String, Object>> entrySet() {
+      return delegate.entrySet();
+    }
+  }
+
+  /**
+   * Default implementation of {@link PrimitiveValueAccessor} which use {@link UnaryFunction} to
+   * conversions.
+   *
+   * @param <T> input type
+   * @param <V> output type
+   */
+  public static class PrimitiveValueAccessorImpl<T, V> implements PrimitiveValueAccessor<T, V> {
+
+    private final UnaryFunction<T, V> valueOfConvertCallback;
+    private final UnaryFunction<V, T> createFromConvertCallback;
+
+    public PrimitiveValueAccessorImpl(
+        UnaryFunction<T, V> valueOfConvertCallback, UnaryFunction<V, T> createFromConvertCallback) {
+      Preconditions.checkNotNull(valueOfConvertCallback, "ValueOfConvertCallback can not be null.");
+      Preconditions.checkNotNull(
+          createFromConvertCallback, "CreateFromConvertCallback can not be null.");
+      this.valueOfConvertCallback = valueOfConvertCallback;
+      this.createFromConvertCallback = createFromConvertCallback;
+    }
+
+    @Override
+    public V valueOf(T object) {
+      return valueOfConvertCallback.apply(object);
+    }
+
+    @Override
+    public T createFrom(V object) {
+      return createFromConvertCallback.apply(object);
+    }
+
+    @Override
+    public boolean equals(Object o) {
+      if (this == o) {
+        return true;
+      }
+      return o != null && getClass() == o.getClass();
```

Comment:
What is the reasoning for this equals implementation?

If we don't need this, I'd rather see it removed.

Reply:
removed.